src/PrestaShopBundle/Controller/Admin/CategoryController.php

```diff
@@ -49,6 +50,8 @@ class CategoryController extends FrameworkBundleAdminController
     /**
      * Process Ajax Form to add a simple category (name and parent category).
      *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller')) || is_granted('update', request.get('_legacy_controller'))")
```

Comment:
If I am not wrong, this route was used by Product Page V1 to create categories from inside the product page. There is a button "Create a category", it's a shortcut to avoid asking the user to go to "Create a category" backoffice page, it allows user to do it without leaving the product page. As this controller has been deprecated in 8.1 I guess we don't really care if it's 100% OK.

Reply:
Same comment as above, deprecated is not removed In 9.0 we don't care as it will be removed but in 8.1 it should have as much attention as the other controllers And I wonder if we should we should also add `is_granted('update', request.get('AdminProducts'))` 🤔 But maybe not, permission to edit product doesn't mean you're allowed to edit/create categories

Reply:
When you browse Product Page V1, you have a small form and a button to allow you to create a category, without having to visit the category BO page. So it's an implicit grant of access.

>  But maybe not, permission to edit product doesn't mean you're allowed to edit/create categories
The Backoffice Product page v1 design obliges us to do this.